buildSrc/src/main/java/com/google/firebase/gradle/ReleaseGenerator.kt

```diff
@@ -19,53 +19,50 @@ open class ReleaseGenerator : DefaultTask() {
         val pastRelease = project.property("pastRelease").toString()
         val rootDir = project.rootDir
         val availableModules = parseSubProjects(rootDir)
+        val firebaseLibraries = extractLibraries(availableModules)
 
-        val repo = initRepo(rootDir)
+        val repo = Git.open(rootDir)
         val headRef = repo.repository.resolve(Constants.HEAD)
         val branchRef = getObjectRefForBranchName(repo, pastRelease)
 
-        val changedDirs = getChangedLibraries(repo, branchRef, headRef, FIREBASE_LIBRARIES)
-        val changedModules = findChangedModules(changedDirs, availableModules)
-        writeReleaseConfig(rootDir, changedModules, currentRelease)
+        val changedDirs = getChangedLibraries(repo, branchRef, headRef, firebaseLibraries)
+        writeReleaseConfig(rootDir, changedDirs, currentRelease)
     }
 
-    private fun findChangedModules(changedDirs: List<String>, availableModules: Set<String>): List<String> = changedDirs.flatMap { directory ->
-        val moduleName = directory.replace("/", ":")
-        val ktxModuleName = "$moduleName:ktx"
-        listOf(moduleName, ktxModuleName).filter { availableModules.contains(it) }
-    }
+    private fun extractLibraries(availableModules: Set<String>): List<FirebaseLibrary> {
+        val nonKtxModules = availableModules.filter { !it.endsWith("ktx") }.toSet()
+        return nonKtxModules
+                .filter { possibleModule -> !nonKtxModules.any { it.startsWith("$possibleModule:") } }
```

Comment:
Not everyday you get to see a good use of `any`. Great job!

Reply:
ha, thanks! unfortunately i had to get rid of it. feelsbadman.